File: 백준/Silver/10845. 큐/큐.cc
#include <bits/stdc++.h>
using namespace std;
queue<int> q;
int n;
string s;
int main() {
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);
    cin >> n;
    while(n--) {
        cin >> s;
        if(s == "push") {
            int k;
            cin >> k;
            q.push(k);
        }
        else if(s == "front") {
            if(!q.empty()) cout << q.front() << '\n';
            else cout << -1 << '\n';
        }
        else if(s == "back") {
            if(!q.empty()) cout << q.back() << '\n';
            else cout << -1 << '\n';
        }
        else if(s == "pop") {
            if(!q.empty()) {
                cout << q.front() << '\n';
                q.pop();
            }
            else cout << -1 << '\n';
        }
        else if(s == "size") {
            cout << q.size() << '\n';
        }
        else if(s == "empty") {
            if(!q.empty()) cout << 0 << '\n';
            else cout << 1 << '\n';
        }
    }
}

File: 백준/Silver/10845. 큐/큐.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int[] stack = new int[10004];
        int head = 0;
        int tail = 0;
        int n = Integer.parseInt(br.readLine());

        while (n-- > 0) {
            String[] s = br.readLine().split(" ");

            if (s[0].equals("push")) {
                stack[tail++] = Integer.parseInt(s[1]);
            } else if (s[0].equals("pop")) {
                if (tail - head != 0) {
                    System.out.println(stack[head++]);
                } else {
                    System.out.println(-1);
                }
            } else if (s[0].equals("size")) {
                System.out.println(tail - head);
            } else if (s[0].equals("empty")) {
                if (tail - head == 0) {
                    System.out.println(1);
                } else {
                    System.out.println(0);
                }
            } else if (s[0].equals("front")) {
                if (tail - head != 0) {
                    System.out.println(stack[head]);
                } else {
                    System.out.println(-1);
                }
            } else if (s[0].equals("back")) {
                if (tail - head != 0) {
                    System.out.println(stack[tail - 1]);
                } else{
                    System.out.println(-1);
                    
                }
            }

        }
    }
}